### DISCUSSION TITLE
Script crash when trying to make plot png

### DISCUSSION THREAD

**Discussion by jaloji** [R3S3t9999/DoVi_Scripts#23] [General]
Hi,
I'm trying to make an hybrid from DV P5 WEB to HDR10/HDR10+ from WEB and for that like you did in your tutorial but with bluray, I would like to compare brightness with 6-2 plot and 6-5 plot. I've downloaded every things needed.

But when I'm trying to plot HDR10 file Idoesn't have png file.
When I'm trying to plot DV P5 file the script crash.

By the way if you have some advice to me for what I'm trying to do, you're welcome.

Thanks in advance

**Comment by R3S3t9999** [comment]
Read the instructions on what you have to install and how to create your script folder on the [README](https://github.com/R3S3t9999/DoVi_Scripts).

**Reply by jaloji** [reply]
But I followed your instruction and I've downloaded your tool folder too...

**Comment by R3S3t9999** [comment]
Do you have space or special characters in your folder path? I'm a bit tired of repeating this so I'll do an installation video tutorial tonight when I get home.

**Reply by jaloji** [reply]
No no, as I said I've followed your instructions

**Comment by jaloji** [comment]
I've succefully make the png plot for the HDR10 file but still crash before output a png file for the DV plot and I don't understand why, if something is missing or not, if its me who are doing something wrong or not.

![ahsoka_s01e01_HDR10_plot1](https://github.com/R3S3t9999/DoVi_Scripts/assets/83598824/b9cd63e0-0acb-4959-ba88-f4b2cb6186db)

EDIT: Finally I just found the problem when rewatching again your video tutorial and I made a mistake with the choice of DV measerement (I thank that Level was for DV Profile but it seems that not) so I obtained this one. Do you think that the hybrid is still valuable ?

![ahsoka_s01e01_DoVi_L1_PLOT](https://github.com/R3S3t9999/DoVi_Scripts/assets/83598824/35c7033a-e214-4f3f-9545-78e28af2ab0e)

**Comment by R3S3t9999** [comment]
I dont know but by the look of your HDR plot, it looks like you did not install Python.

INSTALLATION TUTORIAL: https://www.youtube.com/watch?v=6SLtsVDse2w

**Reply by jaloji** [reply]
I've Python installed and I've Python inside tool directory too...

**Comment by R3S3t9999** [comment]
Try to install the version in Microsoft store.
Your HDR plot is supposed to look like this:

![Spring Breakers 2012 2160p WEB_HDR_plot](https://github.com/R3S3t9999/DoVi_Scripts/assets/146230929/dcea89c4-02a2-4ddf-99bb-8c8faf0f037a)

**Reply by jaloji** [reply]
I'm in this cas I will be forced to give up, because I can't reinstall with the systems, I need portable version for this tool, like madvr in your vidéo, I can't destroy all my configuration to read these files with MPC-HC just to try to make hybrid who I don't know if I really can do this by the way...

**Comment by R3S3t9999** [comment]
what lol? Destroy??? The script installation does not destroy anything...
 If you already installed madvr you don't have to install it again.

**Reply by jaloji** [reply]
Yeah I've most of every tools need who are actualy neeeded already installed but when I'have tried to make this work for the first time it didn't recognize any of the dir modification I've made inside the .bat and tell me that this .exe is missing etc... so I've download your tools dire directly and thats like this I've been able to make the two png who I've send to you here...

**Reply by jaloji** [reply]
I've installed an other version of python(the last 3.12) it is in environnement variables so like the first I've had (3.9) there are no problem to call it from any windows cmd, I've modified the lavfilter settings even I didn't want to touch it just to test if I can make your script work or not and again if I make 6-2 and put the HDR file in it I obtain the exact same png so... Either I really don't have the choice to modify all my settings for the same apps used by the script (even if I have already started with LAVFilter), or clearly I'm giving up, I don't understand how it's possible that it crashes at this point when all I want is to compare the two PNGs to know if it is possible to do the hybrid or not.

EDIT: There is now way to debug ? Because if there is any of errors messages I just can't read them, the cmd close immediately after generate png and it doesn't delete the temp folder too

**Reply by jaloji** [reply]
Ok I finally got it works ! And the problem was to launch your script with admin privilege and I don't know why but it solve it so here is the result: 
![ahsoka_s01e01_HDR_plot](https://github.com/R3S3t9999/DoVi_Scripts/assets/83598824/0434b949-5491-4080-9a65-c2102324c4d6)

![ahsoka_s01e01_DoVi_L1_PLOT](https://github.com/R3S3t9999/DoVi_Scripts/assets/83598824/d64eb857-d905-488c-ac53-bb15a4af2422)

**Reply by jaloji** [reply]
Ok so this is no longer a purely technical question but now how can I determine whether or not I can make a clean hybrid? And what about the ICtCp colorspace conversion from DV P5 to YCBCR colorspace for HDR10 with DV P8? Is there any impact or loss of quality? Or only the brightness comparaison is necessary ?

**Comment by R3S3t9999** [comment]
It's all explained in the video. 
1- you compare the graph 
2- you do screenshots comparisons

if both match then the hybrid is ok. All the DV profiles have the same metadata levels. There's no loss in the dynamic metadata when you convert it to P8.

**Reply by jaloji** [reply]
Yes, I've seen that but my question is more about what is important to be careful exactly with the graph comparaison ? And I don't speak about metadata but about the base colorspace used to applied this metadata on which is not the same between HDR and DV P5

**Comment by R3S3t9999** [comment]
of course, the dovi_tool.exe converts the P5  to P8 properly, it adjusts the headers, profile and the reshaping metadata... 
You only have to worry about the frame sync and if the grades match.

**Reply by jaloji** [reply]
Ok I see, so if I follow your advice with the screenshot as exemple I've put here, for you if the screen comparaison with the two video is the same it's ok ? I mean this king of graph difference is not a problem ? So the problem is only when we have DV with more nits than with the HDR ?

**Comment by R3S3t9999** [comment]
Yes, your max_pq peaks look close enough. 

PS: I probably already have checked most of the hybrid [here](https://docs.google.com/spreadsheets/d/1jBIGF8XTVi9VmDBZ8a5hEyongYMCDlUiLHU9n1f_S74/edit#gid=0)

**Reply by jaloji** [reply]
It's a TV show the source are WEB for the two version HDR and DV so thank I've seen all your google sheet but I don't think it is in ;)

Just a last question, if the first file (WEB again not BR) is HDR10+ we do the same process ?

**Comment by R3S3t9999** [comment]
I've never seen a TV show with different grades in DV and HDR10.  
HDR10plus has its own metadata, so it doesn't matter to DV. The HDR10 hevc stream can have both.

FYI, you can batch-convert to p8 TV show seasons with 1-2.

**Reply by jaloji** [reply]
Oh ! Ok I see so if I want to make an hybrid from WEB and the HDR10 have HDR10+ too, I doen't have to compare with HDR10+ but with HDR10 like right know ?

So if I understand what you said, every tv show from web can be made in hybrid P8 to have just one file ?

**Comment by R3S3t9999** [comment]
you only have to check the grade when you inject metadata(DV or HDR10plus) from different sources.
It is safe to assume that most of the P8 hybrid from the TV shows are correct but the profile 5 web-dl are still better because P5 uses a more efficient colorspace which is the equivalent to 11.5bit YCBCR..

**Reply by jaloji** [reply]
Ok so this what I've asked before and dovi_tool can't convert this "11.5bit YCBCR".. so there is still a loss is this case, or we are talking about somethiing whos are not visible by human eye maybe ?

**Comment by R3S3t9999** [comment]
Your TV is only 10bits.

**Reply by jaloji** [reply]
Oh yeah of course ! Ok thank you for everything and particulary for your time !

**Comment by jaloji** [comment]
Hello,
I just realized that I forgot to ask you one last question, sorry.

I was able to observe that there was a general tendency for the DV P5 file to have a better bitrate than the HDR10 file (differences of up to 2-3GB and this just because of the difference in bitrate), in still talking about the principle that both are WEB.

And so according to your experience, after having injected the DV metadata into the HDR file as we have not touched the bitrate significantly, does this have a direct impact on the final rendering or not? Because that leaves us to assume that most of the additional bitrate is well used for the equivalent of HDR10 in DV version, namely the famous "11.5bit YCBCR", but I doubt that such significant differences are caused only for the additional 1.5 bit .

**Comment by R3S3t9999** [comment]
The bitrate is important but the only way to really know if one encode is better than another one is by making screenshot comparisons (workflow 7-2). For example, even though they have the same bitrate, I find the P5 Movie anywhere webdl to have much lower quality than the HDR10 encode. On iTunes, the quality/bitrate is the same but their P5 encodes have static reshaping metadata that messes up the color so I always prefer P8 hybrid for itunes/APTV.  On Netflix/Disney/Amazon, the P5 encode is usually better encoded. You have [some examples here](https://docs.google.com/spreadsheets/d/1jBIGF8XTVi9VmDBZ8a5hEyongYMCDlUiLHU9n1f_S74/edit#gid=917134052)

P5 webdl are 10bits, not 11.5bit.  But because ICtCp is more efficient, it's been calculated that it was the equivalent of 11.5bits YCbCr.

![p5 itp colorspace](https://github.com/R3S3t9999/DoVi_Scripts/assets/146230929/7e07f43c-c42a-40a6-819e-d9022b8a5f12)

**Reply by jaloji** [reply]
Ok but but if the P5 encode have higher bitrate and screenshot comparaison show a better encode (even I think this is very subjective method) what we can do ? Just don't do the hybrid ?

**Comment by R3S3t9999** [comment]
Screenshot comparisons are NOT subjective as long as you know what to look for. 
If the quality is the same, I go for the P5 unless it's from iTunes.

**Reply by jaloji** [reply]
Ok, so I don't know what to look for, I'm not an expert on this subject

**Reply by jaloji** [reply]
Ok so here is an example of brightness too different (bitrate is same)

**Comment by R3S3t9999** [comment]
No it's close enough... Small differences are acceptable since the DV metadata can be slightly manipulated by the colorist which is why you should always make a screenshot comparison and you will see that it's the same.

when the brightness is not the same, it will be obvious: https://slow.pics/c/BKMEbFX4

**Reply by jaloji** [reply]
Ohhhh That kind of difference ! Okay its more clear for me ! Thanks again

**Comment by jaloji** [comment]
Hello,
just found an exemple of "impossible hybrid", I think:
![Tehran S02E01 FRENCH CUSTOM DV 2160p WEB H265 EAC3_DoVi_L1_PLOT](https://github.com/R3S3t9999/DoVi_Scripts/assets/83598824/885bf276-65b1-4b34-849e-6a34788b5f55)
![tehran s02e01 multi hdr 2160p web h265-mcnulty_HDR_plot](https://github.com/R3S3t9999/DoVi_Scripts/assets/83598824/ed871193-2db4-4406-a25a-8e9938064533)

**Reply by R3S3t9999** [reply]
not necessary, as I said, L1 can be manipulated by the colorist while the HDR10 plot with madvr is the actual measurement of the HDR grade brightness. Do not only rely on plots and **always do screenshot comparisons.**
